### ISSUE TITLE
Relationship with existing scikit-learn IR methods

### ISSUE THREAD

**Issue by rth** [johannfaouzi/pyts#5]
Just a few thoughts from quickly running the examples, regarding the relationship of some of these estimators to those of scikit-learn.

 Maybe mentioning some of it in the docstrings could be useful for users familiar with scikit-learn,

* `pyts.transformer.StandardScaler` is similar to `sklearn.preprocessing.StandardScaler` except that the scaling is done along axis=1 instead of 0, (i.e. equivalent of using `sklearn.preprocessing.scale(X, axis=1)`) with an additional "empirical variance" added during the normalization.
 
* For `VSM(window_size=4, numerosity_reduction=False)` + `SAXVSMClassifier`, I wonder if it's equivalent to running
   `CountVectorizer(analyzer='char', ngram_range=(4, 4))` on the SAX output, followed by class-wise normalization with `TfidfTransformer` and finally applying the `NearestCentroid(metric='cosine')`. I've not read the SAX-VSM paper in detail, so I might be missing something.

In any case these are interesting analogies, and something that might be used for unit tests #3  (if true). Also using the scikit-learns TF-IDF methods, that have been optimized for a while, will probably be computational faster (in particular they use sparse arrays instead of lists).

 I know you don't plan to actively develop this package at the moment, so just writing it for future reference..

**Comment by johannfaouzi** [comment]
I'm glad to read your thoughts: I'm quite excited to work a bit on this project!

There are a lot of functions/classes from numpy and scikit-learn that I don't know and that could have helped me quite a bit probably. I'm going to check if it's equivalent. Thanks for making me discover new useful functions/classes.

**Comment by johannfaouzi** [comment]
I looked a bit at the aforementioned functions from scikit-learn and it kinda does what I want. 

However, in the SAX-VSM paper, they "merge" all documents belonging to the same class into a single document (which means that the number of documents before applying tf-idf is equal to the number of classes, regardless of the number of samples).

For this reason, I would rather keep my own function for `VSM` and use `TfidfVectorizer(analyser='word')`.

At the end, for the prediction phase, there is no need for `NearestCentroid` because the "pooling" is done before tf-idf and not after. `cosine_similarity` does the trick.

The next version of the package will include these changes (as well as small other improvements and some unit tests). I hope I can push it within the next month.

**Comment by rth** [comment]
Yes, SAX-VSM does something quite specific, but I just wanted to make you aware of those methods to see if there is anything that can be reused from there.
